disk.go

```diff
@@ -84,6 +85,55 @@ func (d *Disk) NowIsDone() bool {
 	return true
 }
 
+//==========================================================================
+// Customizable Latency Simulation framework for Disk
+//==========================================================================
+
+// This structure defines the possible parameters that influences the latency
+// function for Disk. Its only queue size now, but will grow in future.
+type DiskLatencyParams struct {
+	qSize	float64
+}
+
+// Interface which implements the latency function
+type DiskLatencySimulator interface {
+	GetName() string
+	Latency(params DiskLatencyParams) time.Duration
+}
+
+type LatencyNil string
+
```

Comment:
nitpicking: I don't like this part, starting latencyNil and the next ~10 lines until the Register...() function. 

1) The latencyNil must be a struct type that implements the DiskLatencySimulator interface. Not sure why it is a string. 
2) All static declarations must probably move into a single inited struct. 
3) You can always return 0 instead of duration, no need to multiply it. 
4) Golang map can always grow, s/maxLatencySimulators/initialLatencySimulators/
5) Latency(params DiskLatencyParams) time.Duration could be variadic: (args ...interface{}) or at least (args ...float64)
6) Makes sense to add parabolic latency into the code, alongside latency-nil